src/discof/tower/fd_tower_tile.h

```diff
@@ -1,3 +1,42 @@
+#ifndef HEADER_fd_src_discof_tower_fd_tower_tile_h
+#define HEADER_fd_src_discof_tower_fd_tower_tile_h
+
 #include "../../disco/topo/fd_topo.h"
 
+/* In response to finishing replay of a slot, the tower tile will
+   generate an update to the vote state, and potentially advance
+   the root. */
+
+struct fd_tower_slot_done {
+   /* The reset slot is always set and represents the current fork to
+      build on, it may be unchanged since the last slot done. */
+  ulong reset_slot;
+
+  /* Sometimes, finshing replay of a slot may cause a new slot to be
+     rooted.  If this happens, new root will be 1 and both root_slot and
+     root_block_id will be set to the new root values accordingly.
+     Otherwise, new root will be 0 and root_slot, root_block_id have
+     undefined values. */
+  int new_root;
+  ulong root_slot;
+  fd_hash_t root_block_id;
```

Comment:
```suggestion
  /* Sometimes, finishing replay of a slot may cause a new slot to be
     rooted.  If this happens, new root will be 1 and both root_slot and
     root_block_id will be set to the new root values accordingly.
     Otherwise, new root will be 0 and root_slot, root_block_id have
     undefined values. */

  int       new_root;
  ulong     root_slot;
  fd_hash_t root_block_id;

  fd_hash_t block_id;
```